### PR TITLE
check internet

### PR DESCRIPTION
**Pull request by hungvv-1301**
## Related Tickets
- [#Feature #54689: Check Internet](https://edu-redmine.sun-asterisk.vn/issues/54689)
- [#Task #54690: Check Internet](https://edu-redmine.sun-asterisk.vn/issues/54690)
## WHAT
- Create feature check internet
- Fix update layout when confirm bill
- Swipe refresh 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/atom/android/bookshop/ui/BookApplication.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.atom.android.bookshop.ui
+
+import android.app.Application
+import com.atom.android.bookshop.utils.registerNetwork
+
+class BookApplication : Application() {
+    override fun onCreate() {
```
Comment (hungld-1302) [inline app/src/main/java/com/atom/android/bookshop/ui/BookApplication.kt]: cách 1 dòng

File: app/src/main/res/layout/fragment_bill_confirm.xml [hunk 2]
```diff
@@ -7,45 +7,51 @@
...
-            android:layout_height="wrap_content"
-            android:layout_centerHorizontal="true"
-            android:layout_marginTop="@dimen/margin_8dp"
-            android:visibility="gone"
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent" />
-    </androidx.constraintlayout.widget.ConstraintLayout>
+            <TextView
+                android:id="@+id/text_view_get_bill_failed"
```
Comment (hungld-1302) [inline app/src/main/res/layout/fragment_bill_confirm.xml:31]: text thôi

File: app/src/main/res/layout/fragment_bill_delivery.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                android:layout_height="@dimen/_0dp"
+                app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+                app:layout_constraintBottom_toTopOf="@id/progress_loading_more"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent"
+                tools:listitem="@layout/item_bill_delivery" />
+
+            <TextView
+                android:id="@+id/text_view_get_bill_failed"
```
Comment (hungld-1302) [inline app/src/main/res/layout/fragment_bill_delivery.xml:30]: .

File: app/src/main/res/layout/fragment_bill_pending.xml [hunk 4]
```diff
@@ -6,45 +6,50 @@
...
-            android:layout_height="wrap_content"
-            android:layout_centerHorizontal="true"
-            android:layout_marginTop="@dimen/margin_8dp"
-            android:visibility="gone"
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent" />
-    </androidx.constraintlayout.widget.ConstraintLayout>
+            <TextView
+                android:id="@+id/text_view_get_bill_failed"
```
Comment (hungld-1302) [inline app/src/main/res/layout/fragment_bill_pending.xml:29]: .

File: app/src/main/res/layout/fragment_bill_success.xml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                android:layout_height="@dimen/_0dp"
+                app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+                app:layout_constraintBottom_toTopOf="@id/progress_loading_more"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent"
+                tools:listitem="@layout/item_bill_success" />
+
+            <TextView
+                android:id="@+id/text_view_get_bill_failed"
```
Comment (hungld-1302) [inline app/src/main/res/layout/fragment_bill_success.xml:30]: .

File: app/src/main/res/layout/item_bill_confirm.xml [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_margin="@dimen/margin_8dp"
+            android:padding="@dimen/margin_12dp">
+
+
+            <TextView
+                android:id="@+id/title_bill"
```
Comment (hungld-1302) [inline app/src/main/res/layout/item_bill_confirm.xml]: thiếu text_

File: app/src/main/res/layout/item_bill_confirm.xml [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+                android:ellipsize="end"
+                android:maxLines="2"
+                android:textSize="@dimen/text_size_16sp"
+                android:textStyle="bold"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent"
+                tools:text="@string/text_some_thing" />
+
+            <TextView
+                android:id="@+id/content_bill"
```
Comment (hungld-1302) [inline app/src/main/res/layout/item_bill_confirm.xml:37]: .

File: app/src/main/res/layout/item_bill_confirm.xml [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+                android:ellipsize="end"
+                android:maxLines="1"
+                android:textSize="@dimen/text_size_16sp"
+                android:textStyle="normal"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/title_bill"
+                tools:text="@string/text_some_thing" />
+
+            <TextView
+                android:id="@+id/text_view_time_confirm"
```
Comment (hungld-1302) [inline app/src/main/res/layout/item_bill_confirm.xml:50]: .

File: app/src/main/res/layout/item_bill_confirm.xml [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+                android:maxLines="1"
+                android:textColor="@color/red"
+                android:textSize="@dimen/text_size_14sp"
+                android:textStyle="italic"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/content_bill"
+                tools:text="@string/text_some_thing" />
+
+            <ImageView
+                android:id="@+id/image_view_wallet"
```
Comment (hungld-1302) [inline app/src/main/res/layout/item_bill_confirm.xml:64]: image_wallet

File: app/src/main/res/layout/item_bill_confirm.xml [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+                android:layout_height="wrap_content"
+                android:layout_marginTop="@dimen/margin_8dp"
+                android:layout_marginBottom="@dimen/margin_6dp"
+                android:src="@drawable/ic_baseline_account_balance_wallet_24"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/text_view_time_confirm" />
+
+            <TextView
+                android:id="@+id/title_total_money"
```
Comment (hungld-1302) [inline app/src/main/res/layout/item_bill_confirm.xml]: text_

File: app/src/main/res/layout/item_bill_confirm.xml [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+                android:paddingBottom="@dimen/margin_8dp"
+                android:text="@string/text_delivery"
+                android:textSize="@dimen/text_size_14sp"
+                android:textStyle="bold"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/text_view_time_confirm" />
+
+            <TextView
+                android:id="@+id/text_view_cancel"
```
Comment (hungld-1302) [inline app/src/main/res/layout/item_bill_confirm.xml:111]: .

File: app/src/main/res/layout/item_bill_delivery.xml [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_margin="@dimen/margin_8dp"
+            android:padding="@dimen/margin_12dp">
+
+
+            <TextView
+                android:id="@+id/title_bill"
```
Comment (hungld-1302) [inline app/src/main/res/layout/item_bill_delivery.xml]: .

File: app/src/main/res/layout/item_bill_success.xml [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_margin="@dimen/margin_8dp"
+            android:padding="@dimen/margin_12dp">
+
+
+            <TextView
+                android:id="@+id/title_bill"
```
Comment (hungld-1302) [inline app/src/main/res/layout/item_bill_success.xml]: .

File: app/src/main/res/layout/item_order_line.xml [hunk 14]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_margin="@dimen/margin_12dp">
+
+    <ImageView
+        android:id="@+id/img_item"
```
Comment (hungld-1302) [inline app/src/main/res/layout/item_order_line.xml]: image_

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**